HW0/P5/P5.txt

```diff
@@ -1 +1,33 @@
 # Your answers here.
+1. 255 s
+   2^7 + 2^6 + ... + 2^0
+
+2. Yes, faster. 34 s
+   2^7/(2^3) + 2^6/(2^3) + ...+ 2^3/(2^3) 
+   + 2^2/(2^3)[==1] + 2^1/(2^3)[==1] + 2^0/(2^3)[==1]
+
+3. 8 s
+   2^7/(inf)[==1] + 2^6/(inf)[==1] + ...+ 2^3/(inf)[==1] 
+   + 2^2/(inf)[==1] + 2^1/(inf)[==1] + 2^0/(inf)[==1]
+
+4. log2(N)
+
+5. 15 s
+   7(communication) + 8(counting)
+
+6. 1 bag case: 272s
+   256(handing) + 1(coummunication) + 7(communication) + 8(counting)
+
+   2 bag case: 143s
+   128(handing) + 0(coummunication) + 7(communication) + 8(counting)
+   
+
+   There is a better strategy. 
+   More employees means more handing&communication time but less counting time.
+   So the problem is a trade-off.
+
+   A better strategy can be like:
```

Comment:
You can actually get this down to 23 seconds if you give out all 256 bags to one cashier, then have her distribute them in parallel (takes 8 seconds), and then have the cashiers add the values up (takes another 15 seconds).